src/validator.directive.js

```diff
@@ -225,6 +228,7 @@
             ctrl.$render();
             if (attrs.messageId) angular.element(document.querySelector('#' + attrs.messageId)).html('');
             else element.next().html('');
+            if ($validationProvider.validCallback) $validationProvider.validCallback(element);
```

Comment:
It works as a great reset here as, when you reset everything is set back to initial state which... is valid.